ml-agents/mlagents/trainers/torch/components/reward_providers/gail_reward_provider.py

```diff
@@ -21,12 +21,13 @@
 class GAILRewardProvider(BaseRewardProvider):
     def __init__(self, specs: BehaviorSpec, settings: GAILSettings) -> None:
         super().__init__(specs, settings)
-        self._ignore_done = True
+        self._ignore_done = False
         self._discriminator_network = DiscriminatorNetwork(specs, settings)
         self._discriminator_network.to(default_device())
         _, self._demo_buffer = demo_to_buffer(
             settings.demo_path, 1, specs
         )  # This is supposed to be the sequence length but we do not have access here
+        self._discriminator_network.encoder.update_normalization(self._demo_buffer)
```

Comment:
How long will this take for large demo sets? Can this become prohibitive and if yes, should we use a sample of the demo buffer ?

Reply:
I replaced this with samples from the expert demos in the update